ds3-contract-comparator-print/src/main/java/com/spectralogic/ds3contractcomparator/print/utils/HtmlRowGeneratorUtils.java

```diff
@@ -0,0 +1,133 @@
+            final Object objectField = field.get(object);
+            if (objectField == null) {
+                return ImmutableList.of();
+            }
+            if (objectField instanceof ImmutableList) {
+                return (ImmutableList<N>) objectField;
+            }
+            throw new IllegalArgumentException("Object should be of type ImmutableList, but was: " + object.getClass().toString());
+        } catch (final IllegalAccessException e) {
+            LOG.error("Could not retrieve list element " + field.getName() + " from object of class "
+                    + object.getClass().toString() + ": " + e.getMessage(), e);
+
+            return ImmutableList.of();
+        }
+    }
+
+    /**
+     * Retrieves the unique property from the Ds3 object
+     */
+    public static <T> String getUniqueProperty(final T object) {
+        if (object == null) {
+            throw new IllegalArgumentException("Cannot get unique property from null object");
+        }
+        //Most Ds3 modules have a name parameter
+        if (PropertyUtils.isReadable(object, "name")) {
+            return "name";
+        }
+        //Used for Ds3ResponseCode
+        if (PropertyUtils.isReadable(object, "code")) {
+            return "code";
+        }
+        //Used for Ds3ResponseType
+        if (PropertyUtils.isReadable(object, "type")) {
+            return "type";
+        }
+        //Unknown unique property
+        LOG.warn("Unknown unique property for class {}", object.getClass().toString());
+        return "";
+    }
+
+    /**
+     * Retrieves the indentation that should be used with the specified field. If the
+     * field is the unique property of an object, then the indent remains the same,
+     * else, if it is non-unique, then the indent is increased by one.
+     */
+    public static <T> int toFieldIndent(final int curIndent, final T object, final Field field) {
+        final String uniqueProperty = getUniqueProperty(object);
+        return field.getName().equals(uniqueProperty) ? curIndent : curIndent + 1;
+    }
+
+    /**
+     * Gets the property value from the specified object. If the property is "name" or "type", then
+     * any path will be removed. If the property does not exist within the object, than an empty
+     * optional is returned.
+     */
+    public static <T> Optional<String> getPropertyValue(final T object, final String property) {
+        if (object == null) {
+            return Optional.empty();
+        }
+        try {
+            final String value = BeanUtils.getSimpleProperty(object, property);
+            if (value == null) {
+                return Optional.empty();
+            }
+            switch (property.toLowerCase()) {
+                case "name":
+                case "type":
+                    if (value.contains(".")) {
+                        return Optional.of(removePath(value));
+                    }
+                    return Optional.of(value);
```

Comment:
I don't think this return statement is needed.  If I'm not mistaken, without a break or return, this should fall into the default: clause and then hit the return.